handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -321,6 +321,19 @@ private void replaceHandler(ChannelHandlerContext ctx, Selection selection) {
         }
     }
 
+    /**
+     * Users may override this method to implement custom behavior. The default
+     * implementation will simply replace {@code this} {@link SniHandler} instance
+     * with the given {@link SslHandler} using the provided {@link SSLEngine}.
+     *
+     * BEWARE: The {@code hostname} argument can be {@code null}.
+     */
+    protected void replaceHandler(ChannelHandlerContext ctx, String hostname,
+            SslContext sslContext, SSLEngine sslEngine) throws Exception {
+
+        ctx.pipeline().replace(this, SslHandler.class.getName(), SslContext.newHandler(sslEngine));
```

Comment:
On that note, would it also make sense to wrap that cause and rethrow it as an `SniException(hostname, sslContext, cause)`?